lib/redux/actions/async.js

```diff
@@ -140,8 +140,12 @@ export function doLogin(creds, opts) {
         memberships: results[2]
       }));
 
-      // detect if a VCA here and redirect to clinic user select screen
+      // detect if a DSA here and redirect to data storage screen
+      if (!personUtils.userHasDSA(results[0].user, results[1], results[2])) {
```

Comment:
Ah! Yes, that **would** be preferred. When I first tried solving this, I did it as part of `retrieveTargetsFromStorage()`, but moved it earlier to short circuit some additional executions. Totally missed that I should have just been able to check `targetUsersForUpload`. 👍